src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,184 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+import duke.task.Tasks;
+
+/**
+ * The Ui interacts with the user..
+ */
+public class Ui {
+    /**
+     * The Scanner to read inputs.
+     */
+    private final Scanner scanner;
```

Comment:
Maybe this code is self-explanatory? There were also many instances which I feel that the variable name is self-explanatory.